COMPILING.md

```diff
@@ -1,85 +1,319 @@
-Instructions for compiling on a Debian-based system. The package names here are
-valid for Ubuntu 12.10 and may or may not work on your system.
+# General Linux Guide
 
-Building instructions, below, always assume you are running them from the
-Cataclysm:DDA source directory.
+To build Cataclysm from source you will need at least a C++ compiler, some basic developer tools, and necessary build dependencies. The exact package names vary greatly from distro to distro, so this part of the guide is intended to give you higher-level understanding of the process.
+
+## Compiler
+
+You have three major choices here: GCC, Clang and MXE.
+
+  * GCC is almost always the default on Linux systems so it's likely you already have it
+  * Clang is usually faster than GCC, so it's worth installing if you plan to keep up with the latest experimentals
+  * MXE is a cross-compiler, so of any importance only if you plan to compile for Windows on your Linux machine (not covered here)
+
+(Note that your distro may have separate packages e.g. `gcc` only includes the C compiler and for C++ you'll need to install `g++`.)
+
+Cataclysm is targeting C++11 standard and that means you'll need a compiler that supports it. You can easily check if your version of `g++` supports C++11 by running:
+
+    $ g++ --std=c++11
+    g++: fatal error: no input files
+    compilation terminated.
+
+If you get a line like:
+
+    g++: error: unrecognized command line option ‘--std=c++11’
+
+This means you'll need a newer version of GCC (`g++`).
+
+The general rule is the newer the compiler the better.
+
+## Tools
+
+Most distros seem to package essential build tools as either a single package (Debian and derivatives have `build-essential`) or a package group (Arch has `base-devel`). You should use the above if available. Otherwise you'll at least need `make` and figure out the missing dependencies as you go.
+
+You should also install `git` and `gettext` (optional if you don't care about localizations).
```

Comment:
I'd rephrase this (as it implies git is optional if you don't care about localizations) -- perhaps include git in the previous line:

> otherwise you'll at least need `make`, and probably `git`, and figure out [...]

This would let us rephrase this line as:

> You may also wish to install `gettext` for localization support, but it is completely optional and you can compile without it.

Another notable mention would be Lua: you can technically compile without it, but you're not getting the full game if you do.